aws/aws-rfdk
# chore: Created sample CDK app that deploys everything in RFDK

### BEFORE MERGING
I would like to do a final end-to-end test once all PR feedback has been addressed to ensure the app still works as intended. Please let me know when this PR looks ready so that I can perform a final test. Once the test is successful, I will get back to you so we can merge this in.

* [x] Do final end-to-end testing

---

### Notes
See the README.md file. Below stuff is now outdated.

---

### Testing
- Deploy all stacks and ensure deployment succeeds
- Look at CloudWatch logs and ensure there are no errors for:
    - `Repository` construct (Deadline Repository installer logs)
    - `RenderQueue` construct (RCS logs)
    - `UBLLicensing` construct (License Forwarder logs)
- Verify that the License Forwarder can communicate with the Render Queue
    - Check in ECS > [cluster] > [service] > Tasks > Click on a task and make sure there aren't any communication errors (**note:** you may see some communication errors that eventually go away. This is likely because the `UBLLicensing` construct isn't set to be dependent on the `RenderQueue`, so it may deploy before the `RenderQueue`, causing the communication issues) 
- Verify that the Worker instance(s) start up without error (look in logs)

### Revisions
- 2: Rebased onto `mainline` and retargeted CR to `mainline`. Added README.md with instructions on how to use the sample app as well as minor cleanup.
- 3: Rebased onto `mainline` again.
- 4: Updated sample app dependencies and cleanup in README.md
- 5: General cleanup and parameterized `UBLLicense`s in `service-tier.ts`

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*
